libs/math/safemath.go

```diff
@@ -0,0 +1,26 @@
+package math
+
+import (
+	"errors"
+	"math"
+)
+
+var ErrOverflowInt = errors.New("integer overflow")
+
+func SafeAddInt32(a, b int32) (int32, error) {
```

Comment:
godoc, please!

(also: did these get copied from somewhere?)

Reply:
adjusted the ones we have for int64 in validator_set.go. will be moving those here in the future. but didn't want to this PR bigger than what it is now